common/nym-lp/src/message.rs

```diff
@@ -0,0 +1,277 @@
+use std::fmt::{self, Display};
+
+// Copyright 2025 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+use bytes::{BufMut, BytesMut};
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+use serde::{Deserialize, Serialize};
+
+/// Data structure for the ClientHello message
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ClientHelloData {
+    /// Client's LP x25519 public key (32 bytes) - derived from Ed25519 key
```

Comment:
if you want to be returning a nice slice in `LpMessage::payload` you might have to unfortunately get rid if the explicit structure in favour of a single [u8; 96] and then just return slices to corresponding bits